src/boot/efi/linux-aarch64.c

```diff
@@ -0,0 +1,161 @@
+/* SPDX-License-Identifier: LGPL-2.1+ */
+
+#include <efi.h>
+#include <efilib.h>
+#include <libfdt.h>
+
+#include "linux.h"
+#include "linux-aarch64.h"
+
+/* DTB table GUID, as defined by UEFI specification 2.9 */
+/* gnu-efi after 3.0.13 should already define this */
+#ifndef EFI_DTB_TABLE_GUID
+#define EFI_DTB_TABLE_GUID \
+    { 0xb1b621d5, 0xf19c, 0x41a5, {0x83, 0x0b, 0xd9, 0x15, 0x2c, 0x69, 0xaa, 0xe0} }
+static EFI_GUID EfiDtbTableGuid = EFI_DTB_TABLE_GUID;
+#endif
+
+/* Create new fdt, either empty or with the content of old_fdt if not null */
+static void *create_new_fdt(void *old_fdt, int fdt_sz) {
+        EFI_STATUS err;
+        void *fdt = (void *) 0xFFFFFFFFUL;
+        int ret;
+
+        err = uefi_call_wrapper(BS->AllocatePages, 4,
+                                AllocateMaxAddress,
+                                EfiACPIReclaimMemory,
+                                EFI_SIZE_TO_PAGES(fdt_sz),
+                                (EFI_PHYSICAL_ADDRESS*)&fdt);
+        if (EFI_ERROR(err)) {
+                Print(L"Cannot allocate when creating fdt\n");
+                return 0;
+        }
+
+        if (old_fdt) {
+                ret = fdt_open_into(old_fdt, fdt, fdt_sz);
+                if (ret != 0) {
+                        Print(L"Error %d when copying fdt\n", ret);
```

Comment:
please use recently added log_error_stall() for log messages. 

can't we output info about the error code?